tracer/src/Datadog.Trace/IAST/Aspects/System/StringAspects.cs

```diff
@@ -272,4 +272,59 @@ public static char[] ToCharArray(string target, int startIndex, int length)
     {
         return StringModuleImpl.OnStringSubSequence(target, startIndex, target.ToCharArray(startIndex, length));
     }
+
+    /// <summary>
+    /// String.ToCharArray aspect
+    /// </summary>
+    /// <param name="target"> string base instance </param>
+    /// <param name="startIndex"> startIndex parameter </param>
+    /// <returns> String.Remove() </returns>
+    [AspectMethodReplace("System.String::Remove(System.Int32)", AspectFilter.StringLiteral_0)]
+    public static string Remove(string target, int startIndex)
+    {
+        string result = target.Remove(startIndex);
+        OnStringRemove(target, result, startIndex, target.Length);
+        return result;
+    }
+
+    /// <summary>
+    /// String.ToCharArray aspect
+    /// </summary>
+    /// <param name="target"> string base instance </param>
+    /// <param name="startIndex"> startIndex parameter </param>
+    /// <param name="count"> count parameter </param>
+    /// <returns> String.Remove() </returns>
+    [AspectMethodReplace("System.String::Remove(System.Int32,System.Int32)", AspectFilter.StringLiteral_0)]
+    public static string Remove(string target, int startIndex, int count)
+    {
+        string result = target.Remove(startIndex, count);
+        OnStringRemove(target, result, startIndex, startIndex + count);
+        return result;
+    }
+
+    /// <summary>
+    /// String.Insert aspect
+    /// </summary>
+    /// <param name="target"> string base instance </param>
+    /// <param name="startIndex"> startIndex parameter </param>
+    /// <param name="value"> value to insert </param>
+    /// <returns> String.Insert() </returns>
+    [AspectMethodReplace("System.String::Insert(System.Int32,System.String)", AspectFilter.StringOptimization)]
+    public static string Insert(string target, int startIndex, string value)
+    {
+        var result = target.Insert(startIndex, value);
```

Comment:
I know it's playing with the customer code, but couldn't we just test if value == string.Empty and in which case return target without doing any tainting?

Reply:
In .Net framework, if we have an empty string in the value, the result string would not be tainted because the framework would create a new instance of the target string, with the same value but a different reference, so we would need to taint it. For the other frameworks, we can just return the result without calling OnStringInsert.

Reply:
I understood. My suggestion was not to call Insert in that case and just return the same ref but that's probably too dangerous

Reply:
Yes, not calling the original method is risky. For this particular case, I am pretty certain that nothing would happen and we could just emulate what the framework library does, but overriding the framework library behavior could potentially lead us to return a different result. For instance, if the behavior of the framework for that method changes in future versions and starts returning new references again, we would be changing the framework insert method behavior without being aware of that.